CardForm/src/main/java/com/braintreepayments/cardform/utils/CardType.java

```diff
@@ -11,6 +11,13 @@
  */
 public enum CardType {
 
+    // order matters here for Maestro since visa regex will detect anything starting with a 4 as Visa
+    MAESTRO("^(493698|5018|5020|5038|5043|5[6-9]|6020|6304|6703|6759|676[1-3])\\d*",
```

Comment:
In almost all cases, a number starting with a 4 will be a visa. Would it be better to instead of changing the priority order to update the regex for visa to not include `493698` ? That way it'll detect it as a visa for `49369` and then switch to Maestro when it gets to `493698`.

Reply:
(we're going to have the same issue with some elo cards)

Reply:
Got it yeah. We could try refactoring the regex a bit. We probably should re-visit some of the card number parsing to see if we can line it up with js / ios better.